src/platform/intel/ace/include/ace/lib/memory.h

```diff
@@ -0,0 +1,83 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Copyright(c) 2022 Intel Corporation. All rights reserved.
+ *
+ * Author: Bartosz Kokoszko <bartoszx.kokoszko@linux.intel.com>
+ */
+
+#ifdef __PLATFORM_LIB_MEMORY_H__
+
+#ifndef __ACE_LIB_MEMORY_H__
+#define __ACE_LIB_MEMORY_H__
+
+#include <sof/common.h>
+#include <sof/lib/cache.h>
+#if !defined(__ASSEMBLER__) && !defined(LINKER)
+#include <sof/lib/cpu.h>
+#endif
+
+#define SRAM_BANK_SIZE			(128 * 1024)
+
+#define EBB_BANKS_IN_SEGMENT		32
```

Comment:
These we shouldn't need anymore with drivers on Zephyr side.

Reply:
The SRAM_BANK_SIZE and EBB_BANKS_IN_SEGMENT values are used in the basefw_hw_config function (base_fw.c)

Reply:
yeap, all of this memory stuff should be based on caps but looks like currently there is no api to get it from zephyr
(I found only static mtl_hpsram_get_bank_count())

Reply:
@abonislawski I added some code to look up bank counts from DTS . it nowadays lives in soc/xtensa/intel_adsp/common/sram.c .. e.g. 
#define PLATFORM_HPSRAM_EBB_COUNT (DT_REG_SIZE(DT_NODELABEL(sram0)) / SRAM_BANK_SIZE)

The SRAM_BANK_SIZE would ideally be in DTS as well, but in practise it is constant for all platforms SOF supports, so defining that locally is ok (at least until we get first platform where it is different ;) ). But the bank count will be different for most of the targets, so this should come from DTS and not be hardcoded on SOF side.

But yeah, it's a good point that basefw_hw_config needs this. I.e. this is just transient need to use XTOS drivers for a while, but rather something SOF needs in the future as well. I.e. we'd need to have proper/stable interfaces to get this information. This is kind of "/proc/cpuinfo" type of information for the application. But most we already have. CPU count is available in DTS,  SRAM size as well, PAGE_SIZE will be defined as CONFIG_MMU_PAGE_SIZE. I.e. I think we have most of the info for basefw_hw_config already now.